pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,174 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
+                                   fill_value=np.nan,
+                                   fill_nighttime=None):
+    '''
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the two series are used to
+    calculate the missing series. After calculation, the series is
+    run through a nighttime routine, where nighttime values are set based on
+    the fill_value and fill_nighttime parameters.
+
+    The "component sum" or "closure" equation relates the three
+    primary irradiance components as follows:
+    .. math::
+       GHI = DHI + DNI * cos(theta_z)
+
+    Parameters
+    ----------
+    solar_zenith : Series
+        Zenith angles in decimal degrees, with datetime index.
+        Angles must be >=0 and <=180. Must have the same datetime index
+        as ghi, dhi, and dni series, when available.
+    ghi : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as dni, dhi, and zenith series, when available.
+    dhi : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dni, and zenith series, when available.
+    dni : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dhi, and zenith series, when available.
+    dni_clear : Series, optional
+        Pandas series of clearsky dni data. Must have the same datetime index
+        as ghi, dhi, dni, and zenith series, when available. See
+        :py:func:`dni` for details.
+    zenith_limit: Float
+        Solar zenith boundary between night and day, in degrees.
+        For calculation of the component sum, solar_zenith is set to 90 where
+        solar_zenith > zenith_limit.
+    fill_value: Float, default np.nan
+        The value that is used to fill in nighttime value
+    fill_nighttime: String, default None
+        Options include 'fill_value', 'equation', None. This is the
+        fill value for nighttime periods.
+        If 'fill_value' is selected, then nighttime values are
+        filled using the fill_value parameter.
+        If 'equation' is used, nighttime fill periods are based on the
+        component sum equation for night (where DNI is 0):
+            GHI = 0 + DHI
+        If None is selected, then the computed nighttime values based on
+        the component sum equation are returned.
+
+    Returns
+    -------
+    Series
+        Pandas series of the calculated values, based on the component sum
+        equation and corrected for nighttime periods.
+    '''
+    component_sum_df = _complete_irradiance(solar_zenith, ghi,
+                                            dhi, dni, dni_clear)
+    if ghi is None:
+        component = 'GHI'
+    elif dhi is None:
+        component = 'DHI'
+    else:
```

Comment:
I'm thinking it would be better as 

```
elif dni is None:

else:
    raise error
```

Reply:
Updated to this in https://github.com/pvlib/pvanalytics/pull/163/commits/9f053f0fe38ab7c2b199f0b34349c2982bad4c36